vendor/github.com/prometheus/prometheus/storage/local/frankenstein.go

```diff
@@ -364,8 +363,9 @@ func (i *Ingester) Stop() {
 }
 
 func (i *Ingester) loop() {
-	defer i.flushAllUsers(true)
+	defer log.Infof("Ingester exited gracefully")
 	defer close(i.done)
+	defer i.flushAllUsers(true)
```

Comment:
With this many defers, one might as well just do one large on to keep it less error-prone:

```go
defer func() {
  i.flushAllUsers(true)
  close(i.done)
  defer log.Infof("Ingester exited gracefully")
}()
```